src/convertToObject.js

```diff
@@ -5,8 +5,38 @@
  *
  * @return {object}
  */
+
 function convertToObject(sourceString) {
-  // write your code here
+  const stylesObject = {};
+
+  if (!sourceString) {
+    return stylesObject;
+  }
+
+  const stringWithoutComments = sourceString.replace(
+    /\/\*[^*]*\*+([^/*][^*]*\*+)*\//g,
+    '',
+  );
+
+  const matches = stringWithoutComments.match(/(?:[^;"']|"[^"]*"|'[^']*')+/g);
+  const cssDeclarations = matches
+    ? matches.map((decl) => decl.trim()).filter((decl) => decl)
+    : [];
+
+  cssDeclarations.forEach((declaration) => {
+    const [property, ...valueParts] = declaration.split(':');
```

Comment:
- You can use `reduce` to generate an object
- `const [key, value] = declaration.split(':');` After the first steps your string should looks like `color: red;`, so you can just split and remove spaces then